task3/src/main/java/com/epam/vsharstuk/Consumer.java

```diff
@@ -0,0 +1,32 @@
+package com.epam.vsharstuk;
+
+import org.apache.log4j.Logger;
+
+public class Consumer extends Thread{
+
+    private ObjectPool pool;
+    private Logger LOG = Logger.getLogger(Producer.class);
+
+    public Consumer(ObjectPool pool) {
+        this.pool = pool;
+    }
+
+    public Consumer(ObjectPool pool, String threadName) {
+        this.pool = pool;
+        this.setName(threadName);
+    }
+
+    public void run() {
+        while (true) {
```

Comment:
It's better to have a thread safe flag, to be able to stop a thread, e.g. `while (isInterrupted())`